src/Compiler/Lambda/OptLambda.sml

```diff
@@ -237,6 +237,54 @@ structure OptLambda: OPT_LAMBDA =
       val f64_greatereq = f64_cmp "greatereq"
     end
 
+    (* Operations on unboxed vectors *)
+
+    local
+      type exp = LambdaExp
+      fun ccall name argtypes restype =
+          CCALLprim {name=name,instances=[],tyvars=[],
+		     Type=ARROWtype(argtypes,[restype])}
+      fun f256_bin opr (x:exp,y:exp) : exp =
+          PRIM(ccall ("__" ^ opr ^ "_f256") [f256Type,f256Type] f256Type, [x,y])
+      val allocVector : exp =
+        let val zero = PRIM(ccall "__real_to_f64" [realType] f64Type, [REAL ("0.0", NONE)])
+             in PRIM(BLOCKF64prim, [zero, zero, zero, zero])
+        end
+    in
+      fun f256_box (x:exp) : exp = PRIM(ccall "__f256_box" [f256Type] stringType,[x])
+      (* let val box = allocVector
+      *      val () = storeVector(x, box)
+      *   in box
+      *)
+      fun f256_box_alloc (x:exp) : exp =
+        let val box = Lvars.newLvar ()
+            val boxVar = VAR {lvar = box, instances = [], regvars = []}
+            val storeExp: exp =
+              LET { pat = [(Lvars.newLvar (), [], unitType)]
+                  , bind = PRIM(ccall "__f256_store" [f256Type, stringType] unitType, [x, boxVar])
+                  , scope = boxVar
+                  }
+         in 
+           LET { pat = [(box, [], stringType)]
+               , bind = allocVector
+               , scope = storeExp
+               }
+        end
```

Comment:
@melsman This is what I came up with. So the box operation after optimization will be this one. Basically equivalent to the pseudocode in the comment